src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -2675,6 +2675,18 @@ internal bool SupportsCompilation([NotNullWhen(false)] out string? reason)
                 return false;
             }
 
+            if (HasCaseInsensitiveBackReferences(this))
```

Comment:
Doesn't putting this here mean we'll never end up using the compiled version of the implementation, since it'll also call SupportsCompilation and see it return false?